shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -241,18 +241,43 @@ \subsection{MIR Transition}
     \vdash \var{\_} \trans{mir}{} \var{\_} \subseteq
     \powerset (\EpochState \times \EpochState)
   \end{equation*}
+  %
+  \emph{Helper function}
+  \begin{align*}
+      & \fun{mirUpdates} \in \InstantaneousRewards \to \StakeCreds \to
+      (\AddrRWD \mapsto \Coin)\times(\AddrRWD \mapsto \Coin)
+      \\
+      & \fun{mirUpdates}~(\var{irReserves},~\var{irTreasury})~\var{stkCreds} =
+      (\var{updateR},~\var{updateT})\\
+      & ~~~~\where \\
+      & ~~~~~~~~~\var{updateR} =
+        \left\{
+        \fun{addr_{rwd}}~\var{hk}\mapsto\var{val}
+        ~\vert~\var{hk}\mapsto\var{val}\in(\dom{stkCreds})\restrictdom\var{irReserves}
+        \right\}
+        \\
+      & ~~~~~~~~~\var{updateT} =
+        \left\{
+        \fun{addr_{rwd}}~\var{hk}\mapsto\var{val}
+        ~\vert~\var{hk}\mapsto\var{val}\in(\dom{stkCreds})\restrictdom\var{irTreasury}
+        \right\}
+        \\
+  \end{align*}
   \caption{MIR transition-system types}
   \label{fig:ts-types:mir}
 \end{figure}
 
 Figure~\ref{fig:rules:mir} defines the MIR state transition.
-The transition first restricts the instantaneous rewards to those which
-correspond to registered stake keys.
-If the reserve pot is large enough to cover the sum of these rewards,
-the corresponding reward accounts are increased by the appropriate amount
-and the reserve pot is proportionally decreased.
-In either case, if the reserve pot was large enough or not,
-we reset the instantaneous rewards back to the empty mapping.
+The transition uses the function $\fun{mirUpdates}$ to restrict
+the two instantaneous rewards mappings to the registered stake keys,
+and transforms the keys of the mappiing to reward accouts.
+
+If the reserve and treasury pots are large enough to cover the sum
```

Comment:
I guess there's a slight ambiguity here, since it's not clear what happens if one but not both is large enough. But this is pretty clear in the actual definitions, so should be fine!